### PR TITLE
Add MSE loss

### LINKED ISSUE #858: MSELoss support

### PR DESCRIPTION
**Pull request by pglusacTT**
Add MSE Loss module

Closes #858

### CODE REVIEW

File: forge/test/mlir/test_loss.py [hunk 1]
```diff
@@ -62,3 +61,32 @@ def test_cross_entropy_loss(prediction_shape):
...
+        (128, 128),
+    ],
+)
+def test_mse_loss(prediction_shape):
+    forge_loss = forge.op.loss.MSELoss("mse_loss", reduction="avg")
+    torch_loss = torch.nn.MSELoss(reduction="mean")
+
+    prediction = torch.randn(prediction_shape, requires_grad=True).to(torch.float32)
+    prediction_forge = forge.tensor.Tensor.create_from_torch(prediction)
+    target = torch.randn(prediction_shape).to(torch.float32)
```
Comment (pilkicTT) [inline forge/test/mlir/test_loss.py:83]: Do we need this `to(torch.float32)` call?

**Review by pilkicTT — APPROVED** [review]

**Review by vladimirjovanovicTT — CHANGES_REQUESTED** [review]

File: forge/test/mlir/test_loss.py [hunk 2]
```diff
@@ -62,3 +61,32 @@ def test_cross_entropy_loss(prediction_shape):
...
+        (2, 2),
+        (3, 5),
+        (32, 32),
+        (33, 127),
+        (128, 20),
+        (128, 128),
+    ],
+)
+def test_mse_loss(prediction_shape):
+    forge_loss = forge.op.loss.MSELoss("mse_loss", reduction="avg")
```
Comment (vladimirjovanovicTT) [inline forge/test/mlir/test_loss.py:78]: We should test all reductions.
Also, nit: it makes sense to me to use same reduction names as torch.

File: forge/forge/op/loss.py [hunk 3]
```diff
@@ -9,6 +10,50 @@
...
+                predictions.ndim() == labels.ndim()
+            ), f"Number of dimensions must match. Got {predictions.ndim()} and {labels.ndim()}."
+            if min_dim is not None:
+                assert (
+                    predictions.ndim() >= min_dim
+                ), f"Number of dimensions must be at least {min_dim}. Got {predictions.ndim()}."
+            if max_dim is not None:
+                assert (
+                    predictions.ndim() <= max_dim
+                ), f"Number of dimensions must be at most {max_dim}. Got {predictions.ndim()}."
```
Comment (vladimirjovanovicTT) [inline forge/forge/op/loss.py:27]: nit: for clarity, add tensor identifiers to the assert prints.
Like "Nr of dimensions of predictions and labels must match", "Nr of dims of predictions tensor must be..."

**Review by vladimirjovanovicTT — APPROVED** [review]
Looks great
